Golem/ActivityLoop.cs

```diff
@@ -47,29 +48,21 @@ public ActivityLoop(HttpClient httpClient, CancellationToken token, ILogger logg
         {
             while (!token.IsCancellationRequested)
             {
-                _logger.LogInformation("Monitoring activities");
+                _logger.LogDebug("Monitoring activities");
                 var now = DateTime.Now;
                 if (newReconnect > now)
                 {
                     await Task.Delay(newReconnect - now);
                 }
                 newReconnect = now + s_reconnectDelay;
-                if (token.IsCancellationRequested)
-                {
-                    token.ThrowIfCancellationRequested();
-                }
+                token.ThrowIfCancellationRequested();
 
                 try
                 {
-                    var stream = await _httpClient.GetStreamAsync("/activity-api/v1/_monitor");
-                    using StreamReader reader = new StreamReader(stream);
-
-                    await foreach (string json in EnumerateMessages(reader, token).WithCancellation(token))
+                    await foreach (var trackingEvent in _yagna.ActivityMonitorStream(token).WithCancellation(token))
```

Comment:
I wonder if this is not too much of cancellation token used in one place:
- the token is passed to the rest call and if cancellation is requested then it should just return
- the same token is used to wrap the task with another task that can be cancelled

I don't remember when `WithCancellation` was first added but it seems to me that it could be removed,